packages/autorest.go/src/gocodemodel/gocodemodel.ts

```diff
@@ -0,0 +1,1252 @@
+    this.methodName = name;
+    this.client = client;
+    this.httpPath = httpPath;
+    this.httpMethod = httpMethod;
+    this.httpStatusCodes = statusCodes;
+    this.naming = naming;
+    this.isPageable = true;
+  }
+}
+
+export class LROPageableMethod implements LROPageableMethod {
+  constructor(name: string, client: Client, httpPath: string, httpMethod: HTTPMethod, statusCodes: Array<number>, naming: MethodNaming) {
+    this.methodName = name;
+    this.client = client;
+    this.httpPath = httpPath;
+    this.httpMethod = httpMethod;
+    this.httpStatusCodes = statusCodes;
+    this.naming = naming;
+    this.isLRO = true;
+    this.isPageable = true;
+  }
+}
+
+export class NextPageMethod implements NextPageMethod {
+  constructor(name: string, client: Client, httpPath: string, httpMethod: HTTPMethod, statusCodes: Array<number>) {
+    this.methodName = name;
+    this.client = client;
+    this.httpPath = httpPath;
+    this.httpMethod = httpMethod;
+    this.httpStatusCodes = statusCodes;
+    this.isNextPageMethod = true;
+  }
+}
+
+export class StructType implements StructType {
+  constructor(name: string) {
+    this.name = name;
+  }
+}
+
+export class StructField implements StructField {
+  constructor(fieldName: string, type: PossibleType, byValue: boolean) {
+    this.fieldName = fieldName;
+    this.type = type;
+    this.byValue = byValue;
+  }
+}
+
+export class ModelType implements ModelType {
+  constructor(name: string, format: ModelFormat, annotations: ModelAnnotations) {
+    this.name = name;
+    this.format = format;
+    this.annotations = annotations;
+  }
+}
+
+export class ModelAnnotations implements ModelAnnotations {
+  constructor(omitSerDe: boolean) {
+    this.omitSerDeMethods = omitSerDe;
+  }
+}
+
+export class ModelField implements ModelField {
+  constructor(name: string, type: PossibleType, byValue: boolean, serializedName: string, annotations: ModelFieldAnnotations) {
+    this.fieldName = name;
+    this.type = type;
+    this.byValue = byValue;
+    this.serializedName = serializedName;
+    this.annotations = annotations;
+  }
+}
+
+export class ModelFieldAnnotations implements ModelFieldAnnotations {
+  constructor(required: boolean, readOnly: boolean, isAddlProps: boolean, isDiscriminator: boolean) {
+    this.required = required;
+    this.readOnly = readOnly;
+    this.isAdditionalProperties = isAddlProps;
+    this.isDiscriminator = isDiscriminator;
+  }
+}
+
+export class PolymorphicType implements PolymorphicType {
+  constructor(name: string, iface: InterfaceType, annotations: ModelAnnotations) {
+    this.name = name;
+    this.interface = iface;
+    this.annotations = annotations;
+  }
+}
+
+export class InterfaceType implements InterfaceType {
+  constructor(name: string, discriminatorField: string) {
+    this.name = name;
+    this.discriminatorField = discriminatorField;
+    // possibleTypes and rootType are required. however, we have a chicken-and-egg
```

Comment:
Should this comment be on the class, so it's more visible?

Reply:
I moved it to the class ctor.